### PR TITLE
postman: init at 5.5.2

### PR DESCRIPTION
**Pull request by xurei**
###### Motivation for this change
New package, latest release

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/web/postman/default.nix [hunk 1]
```diff
@@ -0,0 +1,97 @@
+{ stdenv, lib, gnome2, fetchurl, pkgs, xlibs, udev, makeWrapper }:
+
+stdenv.mkDerivation rec {
+  name = "postman-${version}";
+  version = "5.5.2";
+
+  src = fetchurl {
+    url = "https://dl.pstmn.io/download/version/${version}/linux64";
```
Comment (andir) [inline pkgs/development/web/postman/default.nix]: since you are only fetching the "source" for `x86_64-linux` you should also change `meta.platforms` to `[  "x86_64-linux" ]`.

File: pkgs/development/web/postman/default.nix [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    # Fix missing libudev
+    ln -s ${udev}/lib/systemd/libsystemd-shared.so $out/lib/libudev.so.1
+    wrapProgram $out/share/postman/Postman --prefix LD_LIBRARY_PATH : ${libPath}:$out/lib
+  '';
+
+  postFixup = ''
+    # Create desktop entry
+    mkdir -p $out/share/applications
+    cat > $out/share/applications/postman.desktop << EOF
```
Comment (andir) [inline pkgs/development/web/postman/default.nix]: you might want to lookup `makeDesktopItem`

File: pkgs/development/web/postman/default.nix [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    # fix the path in the desktop file
+    #substituteInPlace \
+    #  $out/share/applications/masterpdfeditor4.desktop \
+    #  --replace /opt/ $out/opt/
+  '';
+
+  preFixup = let
+    libPath = lib.makeLibraryPath [
+      stdenv.cc.cc.lib
```
Comment (andir) [inline pkgs/development/web/postman/default.nix]: Add `pkgs.udev.lib` here and the you should be able to remove the `libudev.so.1` hack below.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -7790,15 +7790,17 @@ with pkgs;
...
   phantomjs2 = libsForQt5.callPackage ../development/tools/phantomjs2 { };
 
-  pmccabe = callPackage ../development/tools/misc/pmccabe { };
-
   pkgconfig = callPackage ../development/tools/misc/pkgconfig {
     fetchurl = fetchurlBoot;
   };
   pkgconfigUpstream = lowPrio (pkgconfig.override { vanilla = true; });
 
+  pmccabe = callPackage ../development/tools/misc/pmccabe { };
```
Comment (andir) [inline pkgs/top-level/all-packages.nix]: if you want to move this can you make it a separate commit?

**Review by andir — CHANGES_REQUESTED** [review]
Thank you for adding postman. I've left some comments but in general it looks good.

**Comment by jokogr** [comment]
* There is already a PR for Postman 5.5.0 (#33608)
* You could rename your commit message to "postman: init at 5.5.2"

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (jokogr) [reply pkgs/top-level/all-packages.nix]: I think there is no need to move this, just leave the existing order or move the postman attribute.

Comment (xurei) [reply pkgs/top-level/all-packages.nix]: Yeah I did that to fixth e alphabetical ordering, but it's not related to the PR. I will remove that change.

**Comment by xurei** [comment]
Changes applied. I also added the Postman license as PR #33608 is doing

File: pkgs/development/web/postman/default.nix (hunk 1 above)
Comment (xurei) [reply pkgs/development/web/postman/default.nix]: Done

File: pkgs/development/web/postman/default.nix (hunk 3 above)
Comment (xurei) [reply pkgs/development/web/postman/default.nix]: Done. Thanks for this one :-)

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -11167,7 +11169,7 @@ with pkgs;
   strigi = callPackage ../development/libraries/strigi { clucene_core = clucene_core_2; };
 
   subdl = callPackage ../applications/video/subdl { };
-
```
Comment (jokogr) [inline pkgs/top-level/all-packages.nix]: Please remove this space, too.

**Comment by jokogr** [comment]
@xurei could you squash all the commits into one?

**Review by andir — APPROVED** [review]

File: pkgs/development/web/postman/default.nix [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${libPath}:$out/share/postman" \
+      $out/share/postman/Postman
+    patchelf --set-rpath "${libPath}" $out/share/postman/libnode.so
+    patchelf --set-rpath "${libPath}" $out/share/postman/libffmpeg.so
+
+    wrapProgram $out/share/postman/Postman --prefix LD_LIBRARY_PATH : ${libPath}
+  '';
+
+  desktopItem = makeDesktopItem {
```
Comment (andir) [inline pkgs/development/web/postman/default.nix]: You still need to actually use the `desktopItem` anywhere in your install phase. Usually that is done like this:

```bash
ln -s ${desktopItem}/share/applications/* $out/share/applications/
```

It might be a bit misleading since often that attribute is defined within the mkDerivation attribute set. In fact that is only possible since the attribute set has the `rec` keyword infront of it and somewhere in the phases `desktopItem` is references as noted before.

File: pkgs/development/web/postman/default.nix [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+  buildPhase = ":";   # nothing to build
+
+  installPhase = ''
+    mkdir -p $out/share/postman
+    cp -R * $out/share/postman
+    mkdir -p $out/bin
+    ln -s $out/share/postman/Postman $out/bin/postman
+
+    # fix the path in the desktop file
```
Comment (andir) [inline pkgs/development/web/postman/default.nix]: you should remove the commented out lines

**Review by andir — CHANGES_REQUESTED** [review]
Address those few comments and squash them together then this should be good to go in.

**Comment by xurei** [comment]
Changes applied and commits squashed.
